lib/resty/wasmx/proxy_wasm.lua

```diff
@@ -67,6 +76,78 @@ local ffi_ops_type = ffi.typeof("ngx_wasm_plan_t *")
 local ffi_pops_type = ffi.typeof("ngx_wasm_plan_t *[1]")
 
 
+local lua_prop_setter
+local lua_prop_getter
+local c_prop_setter
+local c_prop_getter
+
+
+do
+    -- LuaJIT C callbacks are a limited resource:
+    -- We can't create one of these on each request,
+    -- so we prepare these wrappers, which in turn call a Lua function.
+
+    local function store_c_value(r, ckey, cvalue, lvalue, is_const)
+        local value = tostring(lvalue)
+        cvalue.data = value
+        cvalue.len = #value
+
+        return C.ngx_http_wasm_ffi_set_host_property(r, ckey, cvalue,
+                                                     is_const and 1 or 0,
+                                                     1)
+    end
+
+    c_prop_setter = ffi.cast("ngx_wasm_host_prop_fn_t", function(r, ckey,
+                                                                 cvalue)
+        local lkey = ffi_str(ckey.data, ckey.len)
+        local lval = ffi_str(cvalue.data, cvalue.len)
```

Comment:
I've added code to handle this case, and to make the behavior of the `nil` case consistent with const properties as well. I implemented this by adding a negative-cache flag for const-properties on the C side: that was cleaner than trying to encode that case implicitly in the `value`, since empty-strings are 0-length `ngx_str_t` entries as well.